### PR TITLE
ComputeShader: Add a fastMode switch

### PR DESCRIPTION
**Pull request by Popov72**
See https://forum.babylonjs.com/t/i-am-writing-webgpu-ray-tracer-is-anyone-interested-in-this-topic/42188/16

### CODE REVIEW

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Snapshot stored with reference name:
refs/pull/14604/merge

Test environment: 
https://babylonsnapshots.z22.web.core.windows.net/refs/pull/14604/merge/index.html

To test a playground add it to the URL, for example:

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/14604/merge/index.html#WGZLGJ#4600

Links to test babylon tools with this snapshot:

https://playground.babylonjs.com/?snapshot=refs/pull/14604/merge
https://sandbox.babylonjs.com/?snapshot=refs/pull/14604/merge
https://gui.babylonjs.com/?snapshot=refs/pull/14604/merge
https://nme.babylonjs.com/?snapshot=refs/pull/14604/merge

To test the snapshot in the playground with a playground ID add it after the snapshot query string:

https://playground.babylonjs.com/?snapshot=refs/pull/14604/merge#BCU1XR#0

**Review by RaananW — APPROVED** [review]

File: packages/dev/core/src/Engines/WebGPU/Extensions/engine.computeShader.ts [hunk 1]
```diff
@@ -88,7 +88,9 @@ WebGPUEngine.prototype.computeDispatch = function (
         computePass.setBindGroup(i, bindGroup);
     }
 
-    computePass.dispatchWorkgroups(x, y, z);
+    if (x + (y || 0) + (z || 0) > 0) {
```
Comment (sebavan) [inline packages/dev/core/src/Engines/WebGPU/Extensions/engine.computeShader.ts:91]: could you assign a default value of 0 in the function parameters ?

File: packages/dev/core/src/Compute/computeShader.ts [hunk 2]
```diff
@@ -329,56 +336,58 @@ export class ComputeShader {
      * @returns True if the dispatch could be done, else false (meaning either the compute effect or at least one of the bound resources was not ready)
      */
     public dispatch(x: number, y?: number, z?: number): boolean {
-        if (!this.isReady()) {
+        if (!this.fastMode && !this.isReady()) {
```
Comment (sebavan) [inline packages/dev/core/src/Compute/computeShader.ts:339]: maybe move the !isReady in the following block as they both test !fastmode ?

**Review by sebavan — APPROVED** [review]

**Pull request merged**